classes/submission/SubmissionDAO.inc.php

```diff
@@ -340,7 +422,7 @@ function getById($submissionId, $contextId = null, $useCache = false) {
 		if ($contextId) $params[] = (int) $contextId;
 
 		$result = $this->retrieve(
-			'SELECT	s.*, ps.date_published,
+			'SELECT	s.*,
```

Comment:
it would be good to leave this here, in order to be able to set date published for submissions in _fromRow, else the function getDatePublished cannot be used any more.
Thus maybe to get the setting_name = 'datePublished' for the $submissionRevision, if $submissionRevision does not exist than for revision = 1.